src/Dafda/Serializing/PayloadDescriptor.cs

```diff
@@ -25,8 +27,10 @@ public PayloadDescriptor(string messageId, string topicName, string partitionKey
             PartitionKey = partitionKey;
             MessageType = messageType;
             MessageData = messageData;
-            MessageHeaders = messageHeaders;
-        }
+            MessageHeaders = messageHeaders
+                .Where(k => !string.Equals(k.Key, "messageId", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Seems a bit dirty - why do we need to filter out things just passed to the constructor?
Why should `messageId`and `type` not be part of the headers?

Reply:
Good spotted :)
if we add headers that are allready mandatory, then the serialization will break. 
I am only safeguarding what is allready broken.

Reply:
Yeah - dictionary keys needs to be unique :)
Okay - could the knowledge about which fields needs to be filtered out be kept in `DefaultPayloadSerializer` (this seems to define them), maybe in the form of an extension method doing the filtering for you, ie. `FilterOutDefaultHeaders`  or something?

Reply:
Moved to the serializer.